### PR TITLE
Fix Sphinx python docstrings

### PR DESCRIPTION
**Pull request by Roshrini**
## Description ##
Fixes https://github.com/apache/incubator-mxnet/issues/12941, Fixes https://github.com/apache/incubator-mxnet/issues/12936, Fixes https://github.com/apache/incubator-mxnet/issues/13023, Fixes https://github.com/apache/incubator-mxnet/issues/13026, Fixes https://github.com/apache/incubator-mxnet/issues/13008, Fixes https://github.com/apache/incubator-mxnet/issues/13011, Fixes https://github.com/apache/incubator-mxnet/issues/13022, Fixes https://github.com/apache/incubator-mxnet/issues/12914

@aaronmarkham 

You can check the preview here:
http://mxnet-ci-doc.s3-accelerate.dualstack.amazonaws.com/PR-13160/5/index.html

### Essentials ###
Please feel free to remove inapplicable items for your PR.
- [x] Changes are complete (i.e. I finished coding on this PR)
- [x] All changes have test coverage:
- Unit tests are added for small changes to verify correctness (e.g. adding a new operator)
- Nightly tests are added for complicated/long-running ones (e.g. changing distributed kvstore)
- Build tests will be added for build configuration changes (e.g. adding a new build option with NCCL)
- [x] Code is well-documented: 
- For user-facing API changes, API doc string has been updated. 
- For new C++ functions in header files, their functionalities and arguments are documented. 
- For new examples, README.md is added to explain the what the example does, the source of the dataset, expected performance on test set and reference to the original paper if applicable
- Check the API doc at http://mxnet-ci-doc.s3-accelerate.dualstack.amazonaws.com/PR-$PR_ID/$BUILD_ID/index.html
- [x] To the my best knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change

### CODE REVIEW

**Comment by anirudhacharya** [comment]
@mxnet-label-bot [pr-awaiting-review]

File: python/mxnet/visualization.py [hunk 1]
```diff
@@ -213,12 +213,12 @@ def plot_network(symbol, title="plot", save_format='pdf', shape=None, node_attrs
...
-        a dictionary of Graphviz attribute names and values. For example,
+        a dictionary of Graphviz attribute names and values. For example::
             ``node_attrs={"shape":"oval","fixedsize":"false"}``
-            will use oval shape for nodes and allow variable sized nodes in the visualization.
+        will use oval shape for nodes and allow variable sized nodes in the visualization.
     hide_weights: bool, optional
-        If True (default), then inputs with names of form *_weight (corresponding to weight
-        tensors) or *_bias (corresponding to bias vectors) will be hidden for a cleaner
+        If True (default), then inputs with names of form \*_weight (corresponding to weight
+        tensors) or \*_bias (corresponding to bias vectors) will be hidden for a cleaner
```
Comment (frankfliu) [inline python/mxnet/visualization.py]: ``*_bias`` is better than \ escape

File: python/mxnet/visualization.py [hunk 2]
```diff
@@ -213,12 +213,12 @@ def plot_network(symbol, title="plot", save_format='pdf', shape=None, node_attrs
...
         Specifies the attributes for nodes in the generated visualization. `node_attrs` is
-        a dictionary of Graphviz attribute names and values. For example,
+        a dictionary of Graphviz attribute names and values. For example::
             ``node_attrs={"shape":"oval","fixedsize":"false"}``
-            will use oval shape for nodes and allow variable sized nodes in the visualization.
+        will use oval shape for nodes and allow variable sized nodes in the visualization.
     hide_weights: bool, optional
-        If True (default), then inputs with names of form *_weight (corresponding to weight
-        tensors) or *_bias (corresponding to bias vectors) will be hidden for a cleaner
+        If True (default), then inputs with names of form \*_weight (corresponding to weight
```
Comment (frankfliu) [inline python/mxnet/visualization.py]: ``*_weight`` is better than \ escape

File: python/mxnet/visualization.py [hunk 3]
```diff
@@ -213,12 +213,12 @@ def plot_network(symbol, title="plot", save_format='pdf', shape=None, node_attrs
         input symbol names (str) to the corresponding tensor shape (tuple).
     node_attrs: dict, optional
         Specifies the attributes for nodes in the generated visualization. `node_attrs` is
-        a dictionary of Graphviz attribute names and values. For example,
+        a dictionary of Graphviz attribute names and values. For example::
```
Comment (frankfliu) [inline python/mxnet/visualization.py]: 1.  Avoid inline literal quote in code block
2. Add leading and ending line break to code block:

        node_attrs={"shape":"oval","fixedsize":"false"}

File: python/mxnet/recordio.py [hunk 4]
```diff
@@ -37,7 +37,7 @@ class MXRecordIO(object):
     """Reads/writes `RecordIO` data format, supporting sequential read and write.
 
     Example usage:
-    ----------
+    --------------
```
Comment (frankfliu) [inline python/mxnet/recordio.py]: Please change to: Examples,

Example usage is not supported numpy section title.

Comment (Roshrini) [reply python/mxnet/recordio.py]: fixed

File: python/mxnet/visualization.py (hunk 3 above)
Comment (Roshrini) [reply python/mxnet/visualization.py]: fixed

File: python/mxnet/visualization.py (hunk 1 above)
Comment (Roshrini) [reply python/mxnet/visualization.py]: fixed

File: python/mxnet/visualization.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (Roshrini) [reply python/mxnet/visualization.py]: fixed

File: docs/api/python/ndarray/sparse.md [hunk 5]
```diff
@@ -582,7 +582,7 @@ We summarize the interface for each class in the following sections.
     :members: shape, context, dtype, stype, data, indices, indptr, copy, copyto, as_in_context, asscipy, asnumpy, asscalar, astype, tostype, slice, wait_to_read, zeros_like, round, rint, fix, floor, ceil, trunc, sin, tan, arcsin, arctan, degrees, radians, sinh, tanh, arcsinh, arctanh, expm1, log1p, sqrt, square, __neg__, sum, mean, norm, square, __getitem__, __setitem__, check_format, abs, clip, sign
 
 .. autoclass:: mxnet.ndarray.sparse.RowSparseNDArray
-    :members: shape, context, dtype, stype, data, indices, copy, copyto, as_in_context, asnumpy, asscalar, astype, tostype, wait_to_read, zeros_like, round, rint, fix, floor, ceil, trunc, sin, tan, arcsin, arctan, degrees, radians, sinh, tanh, arcsinh, arctanh, expm1, log1p, sqrt, square, __negative__, norm, __getitem__, __setitem__, check_format, retain, abs, clip, sign
```
[the remark this answers is not in the document]
Comment (Roshrini) [reply docs/api/python/ndarray/sparse.md]: There is no function named "__negative__" in sparse.py.. So, it cannot find it's docstring.

**Comment by nswamy** [comment]
@Roshrini could you please fix the conflicts and provide the output that @aaronmarkham is asking.

**Comment by Roshrini** [comment]
Addressed comments and resolved conflicts. @aaronmarkham can you check if this PR is ready to merge?

**Comment by kalyc** [comment]
@mxnet-label-bot update [pr-awaiting-merge]

**Comment by aaronmarkham** [comment]
@mxnet-label-bot add [Doc]

**Pull request merged**